hotkey: Add `Shift` + `V` keyboard shortcut for the read receipts menu.

sayamsamal opened the pull request:
The read receipts option, receides under the triple-dot message actions menu. This made the process of viewing the read receipts take up multiple steps, even via a keyboard-driven workflow.

Via this PR, now while focused on a message in any message feed, and pressing `Shift` + `V`, efficiently brings up the read receipts for that message.

Fixes part of #24716.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
| Screencast |
|--------|
| ![shift+v](https://github.com/zulip/zulip/assets/82862779/0dd730b2-4d14-4b51-bfd5-422a94d5a1e2) |

| Desc | Before | After |
|--------|--------|--------|
| Message actions menu | ![image](https://github.com/zulip/zulip/assets/82862779/07a7e477-0020-4a18-85a5-471e7d525611) | ![image](https://github.com/zulip/zulip/assets/82862779/e5d367c4-7531-4b0c-b6ae-83cc6f3cdc77) |
| `?` menu | ![image](https://github.com/zulip/zulip/assets/82862779/80f24837-af86-4d38-b345-662ab6f4c3ae) | ![image](https://github.com/zulip/zulip/assets/82862779/33a8dcb7-4c7b-43df-bb9d-2e22d1f9cedd) |
| Help center | ![image](https://github.com/zulip/zulip/assets/82862779/d0e567eb-7a24-4605-9c10-4fa38c1342c1) | ![image](https://github.com/zulip/zulip/assets/82862779/d5f8d5d5-8fcc-4e69-b584-a10e914ec132) |
|Fix duplication bug when opening menu repeatedly | ![read-receipts-before](https://github.com/zulip/zulip/assets/82862779/f4b42da8-2c3e-4f57-a6e9-8a692d67c2ea) | ![read-receipts-after](https://github.com/zulip/zulip/assets/82862779/9961ae7a-29d2-4fc5-8e56-077be2366fa9) |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
Thanks! Please include screenshots of the documentation changes.

timabbott commented:
Code changes look reasonable. Do we want it to toggle the modal if its already open? May be worth playing with -- that sort of thing can be annoying if the double-press is unintentional, but nice if it isn't.

alya commented:
I think it's hard to press Shift+V twice by accident, so making it toggle the model sounds like a very nice touch.

alya commented:
In the `?` menu, let's move this way up to be right after "Show message sender's user card".

@timabbott Also suggested making `E` be the # 2 message action on the list, which I agree seems like an improvement, so we can change that while we're here.

And then the help center should have the same ordering as well.

sayamsamal commented:
Changes after feedback:
1. `Shift` + `V` now toggles the read receipts modal.
2. Moved "View read receipts" to show right after "Show message sender's user card" in the message action list of `?` and `/help` 
3. Moved "Edit selected message or view source" (`E`) to `#2` in the message action list of `?` and `/help` 
4. Added the missing screenshots of changes.
5. Added additional tests in `hotkey.test.js`

alya commented:
Thanks!

@timabbott and I chatted a bit about whether to label it as "Toggle read receipts" or "View read receipts" (and forgot to write it down in a comment, sorry!). @timabbott 's proposal was to just  use "View read receipts", since that's the main functionality. I think that sounds right? Thinking about it now, in the help center (where there's plenty of space), we can also add a note saying that the shortcut can be used to close the menu.

alya commented:
If you think it's worth a #design discussion, that would be totally reasonable.

timabbott commented on web/src/hotkey.js:
```diff
@@ -716,6 +718,13 @@ export function process_hotkey(e, hotkey) {
 
     // `list_util` will process the event in send later modal.
     if (modals.any_active() && modals.active_modal() !== "#send_later_modal") {
+        if (
+            event_name === "toggle_read_receipts" &&
+            modals.active_modal() === "#read_receipts_modal"
+        ) {
+            read_receipts.hide_user_list();
+            return true;
+        }
         return false;
     }
```
Let's at least make the outer block `modals.any_active()` and then have a couple `if` blocks inside; `modals.any_active` queries the DOM, so we don't want to do it multiple times if we don't need to.

sayamsamal replied:
I ended up storing the values returned by `modals.any_active()` and `modals.active_modal()` as constants since both of these query the DOM. (`modals.active_modal()` calls `modals.any_active()` inside it's definition, which in turn queries the DOM). It made sense to store it as constants since the active modal and the modal state doesn't change in between the processing of the hotkeys.

timabbott commented:
This lgtm but I did notice if you hit `Shift+V` repeatedly in quick succession, you can get this state:

![image](https://github.com/zulip/zulip/assets/2746074/dee9119d-324f-4c78-ad3a-df5af8c2c150)

Probably worth debugging that while we're working on this. And I like @alya's documentation proposal; one thing in particular about "Toggle read receipts" is that it sounds more like that's going to display them inside messages, not in a modal; so it doesn't set the best expectations for what this feature is.

sayamsamal commented:
I have pushed the following changes:
1. Renamed the option back to "View read receipts", based on the suggestion by @alya and @timabbott.
(The reason behind this change made sense, i.e, the main intention of this option is to show the view read receipts menu, and having it toggle the read receipts is more of a helper feature.)
2. Fixed the read receipts duplication bug when pressing `Shift` + `V` in quick succession. (Tried to explain the changes in detail in the commit message.)

@timabbott this PR is ready for another review.

timabbott commented:
Merged, thanks @sayamsamal!
